### ISSUE TITLE
`fmt::Debug` should not escape printable characters

### ISSUE THREAD

**Issue by pnkfelix** [rust-lang/rust#24588]
(imported from improperly closed bug #22412)

``` rust
pub fn main() {
    println!("{}", "äöüß");
    println!("{:?}", "äöüß");
}
```

prints:

```
äöüß
"\u{e4}\u{f6}\u{fc}\u{df}"
```

but probably should print:

```
äöüß
"äöüß"
```

(See discussion on #22412 for strategies for fixing this; team seemed pretty much on board for this change.)

**Comment by steveklabnik** [comment]
Triage: no change. I do wonder about this kind of change with regards to stability though.

**Comment by estebank** [comment]
I see where the sentiment comes from, but do consider that there're multiple ways to encode the same visual output. `Debug` using the `\u{}` format for those characters helps differentiate easily between `ö` and `o¨`, for example.

**Comment by bluss** [comment]
This was fixed in #34485 and the examples in the original report have the desired output.
